### PR TITLE
docs: table sharding in user guide

### PR DESCRIPTION
**Pull request by waynexia**
## What's Changed in this PR

Closes https://github.com/GreptimeTeam/docs/issues/1046

Describe the when and how to shard a table

## Checklist

- [x] Please confirm that all corresponding versions of the documents have been revised.
- [x] Please ensure that the content in `summary.yml` matches the current document structure when you changed the document structure.
- [x] This change requires follow-up update in localized docs.

## Summary by CodeRabbit

- **Documentation**
  - Added new documentation on table sharding in GreptimeDB, explaining its importance in distributed databases, guidelines for sharding tables, and partition rules based on column value ranges.
  - Updated contributor guide to focus on data sharding concepts, removing specific mentions of partitioning types.
  - Introduced a new section titled `table-sharding` in the summary of topics.

### CODE REVIEW

**Comment by waynexia** [comment]
@coderabbitai pause

**Comment by waynexia** [comment]
@coderabbitai resolve

**Review by zhaoyingnan01 — APPROVED** [review]

File: docs/nightly/en/user-guide/operations/table-sharding.md
Comment (nicecui) [inline docs/nightly/en/user-guide/operations/table-sharding.md:80]:
```suggestion
GreptimeDB provides severals system table to check DB's state. For table sharding information, you can query [`information_schema.partitions`](/reference/sql/information-schema/partitions.md) which gives the detail of partitions inside one table, and [`information_schema.region_peers`](/reference/sql/information-schema/region-peers.md) which gives the runtime distribution of regions.
```

**Review by nicecui — COMMENTED** [review]
LGTM, and when @killme2008 or @fengjiachun is free, it is better to confirm all the description is correct.

File: docs/nightly/en/summary.yml
Comment (killme2008) [inline docs/nightly/en/summary.yml:79]: I think it's not a part of the operation but table management.

File: docs/nightly/en/user-guide/operations/table-sharding.md
Comment (killme2008) [inline docs/nightly/en/user-guide/operations/table-sharding.md:16]: Include a diagram illustrating how tables, regions, frontends, and datanodes collaborate.

File: docs/nightly/en/user-guide/operations/table-sharding.md
Comment (killme2008) [inline docs/nightly/en/user-guide/operations/table-sharding.md:29]: Add a note? The partition columns should be in primary key constraint.

File: docs/nightly/en/user-guide/operations/table-sharding.md
Comment (killme2008) [inline docs/nightly/en/user-guide/operations/table-sharding.md:54]: Better to change the table name.

File: docs/nightly/en/user-guide/operations/table-sharding.md
Comment (killme2008) [inline docs/nightly/en/user-guide/operations/table-sharding.md:80]: check regions state?

**Comment by waynexia** [comment]
hold this for repartition feature

**Comment by nicecui** [comment]
may be we can write a new PR for the repartition feature

**Pull request closed without merging**
